src/utils/azure.ts

```diff
@@ -42,3 +42,29 @@ export const getStorageConnection = async (name: string) =>
   (await execa(
     `az storage account show-connection-string --name ${name} --output tsv`
   )).stdout;
+
+/**
+ * hasCosmosConnection checks if the host has az cli installed and resouceGroup and name
+ * are valid inputs for cosmosdb
+ */
+export const hasCosmosConnection = async (
+  resourceGroup: string,
+  name: string
+) => {
+  try {
+    const successExitCode = "SUCCESS";
+    const documentEndpoint = await execa(
+      `az cosmosdb show -g ${resourceGroup} -n ${name} --query documentEndpoint -o tsv`
+    );
+    const primaryReadonlyMasterKey = await execa(
+      `az cosmosdb list-keys -g ${resourceGroup} -n ${name} --query primaryReadonlyMasterKey -o tsv`
+    );
+
+    return (
+      documentEndpoint.exitCodeName === successExitCode &&
+      primaryReadonlyMasterKey.exitCodeName === successExitCode
+    );
+  } catch (e) {
+    return false;
```

Comment:
it could be useful to log the exception to the console, at least it gives you a hint on why the command failed

Reply:
I used cli.log instead of cli.error because this one causes the test failure